src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5,14 +5,1235 @@
+        }
+
+        if (WI_IsFlagSet(modifiers, Settings::KeyModifiers::Alt))
+        {
+            buffer += L"Alt+";
+        }
+
+        return buffer;
+    }
+
+    // Method Description:
+    // - Takes a MenuFlyoutItem and a corresponding KeyChord value and creates the accelerator for UI display.
+    //   Takes into account a special case for an error condition for a comma
+    // Arguments:
+    // - MenuFlyoutItem that will be displayed, and a KeyChord to map an accelerator
+    void TerminalPage::_SetAcceleratorForMenuItem(Windows::UI::Xaml::Controls::MenuFlyoutItem& menuItem, const winrt::Microsoft::Terminal::Settings::KeyChord& keyChord)
+    {
+#ifdef DEP_MICROSOFT_UI_XAML_708_FIXED
+        // work around https://github.com/microsoft/microsoft-ui-xaml/issues/708 in case of VK_OEM_COMMA
+        if (keyChord.Vkey() != VK_OEM_COMMA)
+        {
+            // use the XAML shortcut to give us the automatic capabilities
+            auto menuShortcut = Windows::UI::Xaml::Input::KeyboardAccelerator{};
+
+            // TODO: Modify this when https://github.com/microsoft/terminal/issues/877 is resolved
+            menuShortcut.Key(static_cast<Windows::System::VirtualKey>(keyChord.Vkey()));
+
+            // inspect the modifiers from the KeyChord and set the flags int he XAML value
+            auto modifiers = AppKeyBindings::ConvertVKModifiers(keyChord.Modifiers());
+
+            // add the modifiers to the shortcut
+            menuShortcut.Modifiers(modifiers);
+
+            // add to the menu
+            menuItem.KeyboardAccelerators().Append(menuShortcut);
+        }
+        else // we've got a comma, so need to just use the alternate method
+#endif
+        {
+            // extract the modifier and key to a nice format
+            auto overrideString = _FormatOverrideShortcutText(keyChord.Modifiers());
+            auto mappedCh = MapVirtualKeyW(keyChord.Vkey(), MAPVK_VK_TO_CHAR);
+            if (mappedCh != 0)
+            {
+                menuItem.KeyboardAcceleratorTextOverride(overrideString + gsl::narrow_cast<wchar_t>(mappedCh));
+            }
+        }
+    }
+
+    // Method Description:
+    // - Show a ContentDialog with a single button to dismiss. Uses the
+    //   FrameworkElements provided as the title and content of this dialog, and
+    //   displays a single button to dismiss.
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens.
+    // Arguments:
+    // - titleElement: the element to use as the title of this ContentDialog
+    // - contentElement: the element to use as the content of this ContentDialog
+    // - closeButtonText: The string to use on the close button
+    fire_and_forget TerminalPage::_ShowDialog(const IInspectable& titleElement,
+                                              const IInspectable& contentElement,
+                                              const winrt::hstring& closeButtonText)
+    {
```

Comment:
Do we maybe want to leave `_ShowDialog` in `App`? We can only have one ContentDialog per application, so it kinda makes sense to leave the `_dialogLock` and such as part of `App` to me. We'd then probably need to have TerminalPage capable of raising some sort of `ShowDialog` event that `App` listens for. I'm open to discussion on this though.

Reply:
Oh, this is also a kind of communication from Page to App. I wonder if you know any ways to achieve this ?

Reply:
Perhaps this should be a static helper in a new UIHelpers or Dialogs class?